services/code_hosting/code_hosting_service.rb

```diff
@@ -18,5 +18,9 @@ def provider_type
     def repos
       not_implemented(__method__)
     end
+
+    def access_to_repo?(repo: nil)
```

Comment:
in this case, I think we actually want it be maybe
`access_to_repo?(repo_config: nil)`
or even 
`access_to_repo?(repo_url: nil)`
The reason being, we have a `GitRepo` object already and that is very heavyweight.

Reply:
changed to repo_url, to be repo type agnostic.